### DISCUSSION TITLE
hidden overloads of ScriptHelper functions?

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#1142] [🙋♀️TE3 Q&A]
Hello, I just checked and the following expression shows as error in Visual Studio
![imatge](https://user-images.githubusercontent.com/40455069/198832020-94912ab4-a6b0-4c00-9162-420af87e82a1.png)
but works fine in Tabular Editor. 

TE3 intellisense does show that there are other overloads... 

is there any way to "unlock" these hidden treasures from Visual Studio? 
hmmm I just checked and is related to the 

`using static TabularEditor.Scripting.ScriptHelper; `

this row allows to use SelecTable without any prefix in visual studio.
if I replace it back to
 
`using TabularEditor.Scripting;`

and use ScriptHelper.SelectTable() then Visual Studio accepts the expression and even gives the correct intellisense message --- 
is there any way to have the best of both worlds?

**Comment by samaguire** [comment]
I did a check in VSCode and it seems to be working, though it is only showing the one method; the one without any tabulareditor types as input. Maybe that's the problem?

![image](https://user-images.githubusercontent.com/62320770/202845409-3596182e-3839-46cb-8fba-0d7e0c6bdaed.png)

![image](https://user-images.githubusercontent.com/62320770/202845449-83d69896-5b3a-4324-8b1d-ff1aa45b57df.png)

**Comment by bernatagulloesbrina** [comment]
That's the exact same situation in Visual Studio then

I guess typing out all the "overloads" needed as methods on top of the code
is the way to go. After all it only needs to be done once.

El sáb, 19 nov 2022 a las 11:02, Stephen Maguire ***@***.***>)
escribió:

> I did a check in VSCode and it seems to be working, though it is only
> showing the one method; the one without any tabulareditor types as input.
> Maybe that's the problem?
>
> [image: image]
> <https://user-images.githubusercontent.com/62320770/202845409-3596182e-3839-46cb-8fba-0d7e0c6bdaed.png>
>
> [image: image]
> <https://user-images.githubusercontent.com/62320770/202845449-83d69896-5b3a-4324-8b1d-ff1aa45b57df.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TabularEditor/TabularEditor3/issues/701#issuecomment-1320849630>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJUUXHIG7CVCJVTGOKASMRTWJCQTPANCNFSM6AAAAAARRY2ZS4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Bernat Agulló

Partner & Data Enthusiast

***@***.***

+34 637 93 42 76
